middlewares/validators/validator.helper.js

```diff
@@ -336,8 +339,60 @@ function jwtValidator (fieldLocation, fieldname, jwtSecret, optional = true) {
 }
 
 /**
- * 
- * @param {"query" | "body" | "header" | "param"} fieldLocation the location where the field should be found 
+ * Validates that field must be a valid search query.
+ * @param {"query" | "body" | "header" | "param"} fieldLocation the location where the field should be found
+ * @param {string} fieldname name of the field that needs to be validated.
+ */
+function searchValidator (fieldLocation, fieldname) {
+    const search = setProperValidationChainBuilder(fieldLocation, fieldname, "Invalid search query");
+    return search.exists().withMessage("Search query must be provided")
+        .custom((value) => {
+            //value is a serialized JSON
+            value = JSON.parse(value);
+            let modelString = query("model", "Corresponding model not found");
+            let model;
+            //Supported models for searching
+            if(modelString.equals('hacker')){
+                model = Models.Hacker;
```

Comment:
So this search is just for Hackers, right

Reply:
Yeah but it's very easy, can add whichever ones